internal/keepers/ocr.go

```diff
@@ -235,14 +235,6 @@ func (k *keepers) ShouldAcceptFinalizedReport(_ context.Context, rt types.Report
 		return false, fmt.Errorf("no ids in report: %s", lCtx)
 	}
 
-	for _, r := range results {
-		// check whether the key is already accepted
-		if k.filter.CheckAlreadyAccepted(r.Key) {
-			k.logger.Printf("%s: key already accepted: %s", r.Key, lCtx.Short())
-			return false, fmt.Errorf("failed to accept key %s as it was already accepted", r.Key)
-		}
-	}
-
```

Comment:
removing this error to have consistent behaviour that we accept all reports we get. If we get it for the same checkBlockNumber it might not result in any update to the filters, but throwing an error can cause issues if there were other upkeeps in batch

Reply:
So this change was because `Accept` avoids changing the filters now? We put this in for the reason of exiting early on report generation if at least one key had already been accepted.

Reply:
I am not sure if we had a strong reason for it to exist in first place. For reports containing multiple upkeeps, if one key was accepted before and one wasn't then it could cause issues, some nodes may accept the report some might not

Having a uniform logic that we accept all reports helps avoid edge cases when a node's local filters can get out of sync

Reply:
I think the problem is that when `Accept` returns an error, we exit building the report. However, since `Accept` changes the filter state, exiting without a report blocks keys from being added to the next report. Keys will end up getting blocked, but not from having a report generated. We won't see the effects of this with one key per report, but it will come back once we start doing multiple per report again.

Reply:
The consequence of that will be more logs with upkeep perform errors, right? What if all keys return an error in `Accept`? We should avoid sending a report at that point.

Reply:
updated should transmit logic per offline discussion https://github.com/smartcontractkit/ocr2keepers/pull/77/commits/7df09b43c6d6e1a7109b0f57c1242e06c63faee8